lib/srv/discovery/common/renaming.go

```diff
@@ -0,0 +1,238 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/services"
+)
+
+// ApplyAWSDatabaseNameSuffix applies the AWS Database Discovery name suffix to
+// the given database.
+// Format: <name>-<matcher type>-<region>-<account ID>.
+func ApplyAWSDatabaseNameSuffix(db types.Database, matcherType string) {
+	if hasOverrideLabel(db, types.AWSDatabaseNameOverrideLabels...) {
+		// don't rewrite manual name override.
+		return
+	}
+	meta := db.GetAWS()
+	suffix := makeAWSDiscoverySuffix(databaseNameValidator,
+		db.GetName(),
+		matcherType,
+		meta.Region,
+		meta.AccountID,
+	)
```

Comment:
ah, thanks for pointing this out!

For Azure: Redis vs Redis Enterprise names must be unique because of the way Azure sets up DNS. I double checked that Azure enforces this just now.
Redis: `<name>.redis.cache.windows.net`
SQL Server: `<name>.database.windows.net`
Postgres: `<name.postgres.database.azure.com`
MySQL: `<name>.mysql.database.azure.com`

After experimenting some more in aws console, within a region database instance name must be unique amongst other instances, and Aurora cluster name must be unique amongst clusters. Since we group these under the single matcher type "rds", it won't be guaranteed to be unique :( I'll have to figure out some way to handle this

Reply:
I believe Redis enterprise uses a different suffix:

<img width="1053" alt="Screenshot 2023-07-22 at 3 13 32 PM" src="https://github.com/gravitational/teleport/assets/12417665/92da4963-815a-4960-a682-4bba9198e698">

https://github.com/gravitational/teleport/blob/2b15263e49da5625922581569834eec4838a9257/api/utils/azure/endpoints.go#L123-L130

But regardless, we may have a problem when multiple fetchers share a matcher.

Reply:
coming back to this:

We are already renaming Aurora cluster dbs with a suffix: `-primary`, `-custom`, or `-reader`, but we use the aurora cluster identifier as the first part of the name, not the individual instances' names.

Therefore, I think we just need to distinguish between cluster dbs and instance dbs to get (most likely) unique naming for RDS/Aurora dbs.
- changed the suffix to use either `rds` or `rds-aurora`.

For Redis, I tested the naming incorrectly in Azure when I tried, you're right that the names can overlap:
- changed the suffix to use either`redis` or `redis-enterprise`

The others are not an issue due to naming uniqueness rules in Azure. Even SQL server vs managed SQL instance.

commit: [09d6033](https://github.com/gravitational/teleport/pull/28845/commits/09d6033ea26c0f4abc60c64267a18ede2b7ef28d)